### PR TITLE
[New] Set max extent

### PR DESCRIPTION
**Pull request by CalebRas**
## Description

This PR implements `Set max extent` in `Maps` category.
URL to README: [https://github.com/Esri/arcgis-maps-sdk-swift-samples/tree/Caleb/New-SetMaxExtent/Shared/Samples/Set%20max%20extent](url)

## Linked Issue(s)

- `swift/issues/4084`

## Screenshots

set-max-extent

### CODE REVIEW

File: Shared/Samples/Set max extent/README.md
Comment (yo1995) [inline Shared/Samples/Set max extent/README.md:1]: Forget to mention: 100 columns page guide only applies to the code comments (2 slashes and 3 slashes), and can be optionally applied to source code (especially when the method call site is too long). All other stuff don't need to be line wrapped, like the README here.

**Review by yo1995 — APPROVED** [review]
After you receive the feedback from another reviewer, you can solve the merge conflict before merging.

**Review by des12437 — APPROVED** [review]
Looks good. The status bar time in the screenshot should be `9:41`, unless it's [not a requirement at this time](https://esri-runtime.slack.com/archives/C0135F8SW8P/p1679092888983559) @yo1995.

To set the status bar time for an iOS 16 simulator I've had success updating the simulator time like this:

cd to the Desktop folder in your terminal and run 
```
xcrun simctl status_bar Booted override --time 9:41 --dataNetwork wifi --wifiMode active --wifiBars 3 --cellularMode active --cellularBars 4 --batteryState charged --batteryLevel 100
```

If that does not work, try 
```
xcrun simctl status_bar "iPhone 14 Pro Max (16.0)" override --time "9:41"
```

**Comment by yo1995** [comment]
Thanks Destiny, I'll let Caleb give it a try. But if he cannot figure that out, I'm OK to forgo that requirement as it may get him into a rabbit hole of setting up multiple versions of simulators and iOS runtime. (also please redact the internal Slack link 😉 )

File: Samples.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -714,6 +719,14 @@
 			path = "Sketch on map";
 			sourceTree = "<group>";
 		};
+		D734FA072A183A5A00246D7E /* Set max extent */ = {
+			isa = PBXGroup;
+			children = (
+				D734FA092A183A5B00246D7E /* SetMaxExtentView.swift */,
+			);
+			path = "Set max extent";
+      sourceTree = "<group>";
```
Comment (yo1995) [inline Samples.xcodeproj/project.pbxproj:728]: The indentation here is off due to that we manually resolved conflicts. To fix this, simply sort the samples by name again, and Xcode will correct this kind of errors.

**Review by yo1995 — APPROVED** [review]

**Review by des12437 — APPROVED** [review]

**Pull request merged**